packages/oslo-generator-shacl-template/lib/OutputHandlerService.ts

```diff
@@ -0,0 +1,50 @@
+import { inject, injectable } from "inversify";
+import { ShaclTemplateGenerationServiceConfiguration } from "@oslo-generator-shacl-template/config/ShaclTemplateGenerationServiceConfiguration";
+import { ShaclTemplateGenerationServiceIdentifier } from "@oslo-generator-shacl-template/config/ShaclTemplateGenerationServiceIdentifier";
+import { QuadStore } from "@oslo-flanders/core";
+import { createWriteStream } from "fs";
+import rdfSerializer from "rdf-serialize";
+import { DataFactory } from 'rdf-data-factory';
+import type * as RDF from '@rdfjs/types';
+import { quadSort } from "./utils/utils";
+
+@injectable()
+export class OutputHandlerService {
+  public readonly config: ShaclTemplateGenerationServiceConfiguration;
+
+  public constructor(
+    @inject(ShaclTemplateGenerationServiceIdentifier.Configuration) config: ShaclTemplateGenerationServiceConfiguration
+  ) {
+    this.config = config;
+  }
+
+  public async write(store: QuadStore): Promise<void> {
+    const df: DataFactory = new DataFactory();
+    let quads = [
+      ...store.findQuads(null, null, null, df.defaultGraph()),
+      ...(store.findQuads(null, null, null, df.namedNode('baseQuadsGraph'))).map(quad => df.quad(quad.subject, quad.predicate, quad.object)),
+    ].sort(quadSort);
+
+    const quadStream = require('streamify-array')(quads);
+    const outputStream = rdfSerializer.serialize(quadStream, { contentType: this.config.outputFormat });
+
+    let fileName: string = this.config.output ? this.config.output : `shacl.${this.getFileExtension()}`;
+    outputStream.pipe(createWriteStream(fileName))
+  }
+
+  private getFileExtension(): string {
+    switch (this.config.outputFormat) {
+      case 'application/ld+json':
```

Comment:
I agree, but maybe this is something we fix in another ticket once we have implemented all the core packages. We could then refactor, because now things change too quickly to have a good overview.

Reply:
Agree! It's beyond the scope of this PR